src/com/squareup/ideaplugin/dagger/handler/ConstructorInjectToProvidesHandler.java

```diff
@@ -45,8 +42,36 @@
   }
 
   private void showUsages(MouseEvent mouseEvent, PsiParameter psiParameter) {
-    new ShowUsagesAction(new Decider.ConstructorParameterInjectDecider(psiParameter)).startFindUsages(
-        PsiConsultantImpl.checkForLazyOrProvider(psiParameter), new RelativePoint(mouseEvent),
-        PsiUtilBase.findEditor(psiParameter), MAX_USAGES);
+    // Check to see if class type of psiParameter has constructor with @Inject. Otherwise, proceed.
+    if (!navigateToConstructorIfProvider(psiParameter)) {
```

Comment:
since `navigateToConstructorIfProvider` potentially performs a UI step I would inverse the boolean and break early. 

```java
if (navigateToConstructorIfProvider(psiParameter)) {
  return;
}
new ShowUsagesAction(...).startFindUsages(...)
```